gadget/ondisk.go

```diff
@@ -85,35 +83,13 @@ type sfdiskPartition struct {
 }
 
 func isCreatedDuringInstall(p *sfdiskPartition, fs *lsblkBlockDevice, sfdiskLabel string) bool {
-	switch sfdiskLabel {
-	case "gpt":
-		// the created partitions use specific GPT GUID types and set a
-		// specific bit in partition attributes
-		if !creationSupported(p.Type) {
-			return false
-		}
-		for _, a := range strings.Fields(p.Attrs) {
-			if !strings.HasPrefix(a, "GUID:") {
-				continue
-			}
-			attrs := strings.Split(a[5:], ",")
-			if strutil.ListContains(attrs, createdPartitionAttr) {
-				return true
-			}
-		}
-	case "dos":
-		// we have no similar type/bit attribute setting for MBR, on top
-		// of that MBR does not support partition names, fall back to
-		// reasonable assumption that only partitions carrying
-		// ubuntu-boot and ubuntu-data labels are created during
-		// install, everything else was part of factory image
-
-		// TODO:UC20 consider using gadget layout information to build a
-		// mapping of partition start offset to label/name
-		createdDuringInstall := []string{ubuntuBootLabel, ubuntuSaveLabel, ubuntuDataLabel}
-		return strutil.ListContains(createdDuringInstall, fs.Label)
-	}
-	return false
+	// we consider structures that have filesystem labels matching those that
+	// are created during install mode to be "created during install", but note
+	// that this is a guess - the caller should only rely on this to do
+	// destructive operations if these on disk structures match up with those in
+	// the gadget.yaml via ensureLayoutCompatibility.
+	createdDuringInstall := []string{ubuntuBootLabel, ubuntuSaveLabel, ubuntuDataLabel}
+	return strutil.ListContains(createdDuringInstall, fs.Label)
```

Comment:
This is nice!

Reply:
for gpt we should use the partition name, no? not the fs label, we might not even have created the fs yet? am I missing something?

Reply:
this no longer uses fs label at all, it only considers structure offsets in relation between the on-disk layout and the gadget layout.